pkg/controllers/allocation/binpacking/packer.go

```diff
@@ -89,20 +91,32 @@ func (p *packer) Pack(ctx context.Context, schedule *scheduling.Schedule, instan
 	// Sort pods in decreasing order by the amount of CPU requested, if
 	// CPU requested is equal compare memory requested.
 	sort.Sort(sort.Reverse(ByResourcesRequested{SortablePods: schedule.Pods}))
+	packs := map[uint64]*Packing{}
 	var packings []*Packing
 	var packing *Packing
 	remainingPods := schedule.Pods
 	for len(remainingPods) > 0 {
 		packables := PackablesFor(ctx, instances, schedule)
 		packing, remainingPods = p.packWithLargestPod(ctx, schedule.Constraints, remainingPods, packables)
 		// checked all instance types and found no packing option
-		if len(packing.Pods) == 0 {
+		if len(flatten(packing.Pods...)) == 0 {
```

Comment:
this might be over thinking things, but seems like flatten might be getting called quite a bit here (up to three times per iteration thru the loop?) Seems like it's only needed to get the length - maybe just use a function that walks thru and counts the length instead?